src/features/Conversations/components/TreeItem/TreeItem.tsx

```diff
@@ -0,0 +1,254 @@
+  childCount?: number;
+  clone?: boolean;
+  collapsed?: boolean;
+  depth: number;
+  disableInteraction?: boolean;
+  disableSelection?: boolean;
+  ghost?: boolean;
+  handleProps?: any;
+  indicator?: boolean;
+  indentationWidth: number;
+  value: string;
+  name: string;
+  type: string;
+  onCollapse?(): void;
+  onRemove?(): void;
+  wrapperRef?(node: HTMLLIElement): void;
+}
+
+export const TreeItem = forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      childCount,
+      clone,
+      depth,
+      disableSelection,
+      disableInteraction,
+      ghost,
+      handleProps,
+      indentationWidth,
+      indicator,
+      collapsed,
+      onCollapse,
+      onRemove,
+      style,
+      value,
+      name,
+      type,
+      wrapperRef,
+      ...props
+    },
+    ref,
+  ) => {
+    const [isDeleting, setIsDeleting] = useState(false);
+    const [isEditing, setIsEditing] = useState(false);
+    const [editedItemName, setEditedItemName] = useState(name);
+    const [itemPlaceholder, setItemPlaceholder] = useState(true);
+
+    const dispatch = useAppDispatch();
+    const { id } = useParams();
+    const navigation = useNavigate();
+
+    const onClickDelete = (event: MouseEvent) => {
+      event.stopPropagation();
+      setIsDeleting(true);
+    };
+
+    const onCancel = (event?: MouseEvent) => {
+      event?.stopPropagation();
+      setIsDeleting(false);
+      setIsEditing(false);
+      if (name !== editedItemName) {
+        setEditedItemName(name);
+      }
+    };
+
+    const onConfirm = (event: MouseEvent) => {
+      event.stopPropagation();
+      if (isDeleting) {
+        onRemove?.();
+        setIsDeleting(false);
+      }
+      if (isEditing) {
+        if (editedItemName) {
+          dispatch(
+            renameChatTreeItem({
+              chatTreeId: value,
+              chatTreeName: editedItemName,
+            }),
+          );
+        }
+        setIsEditing(false);
+      }
+    };
+
+    const onClickEdit = (event: MouseEvent) => {
+      event.stopPropagation();
+      setIsEditing(true);
+    };
+
+    const onChangeName = (event: ChangeEvent<HTMLInputElement>) => {
+      event.stopPropagation();
+      setEditedItemName(event.target.value);
+    };
+
+    const onClickInput = (event: MouseEvent) => {
+      event.stopPropagation();
+    };
+
+    const onOutsideClick = () => {
+      if (isDeleting || isEditing) {
+        onCancel();
+      }
+    };
+
+    const onChatClick = (event: MouseEvent, route: To) => {
+      event.stopPropagation();
+      navigation(route);
+    };
+
+    const onCollapsePlaceholder = (event: MouseEvent) => {
+      event.stopPropagation();
+      setItemPlaceholder(!itemPlaceholder);
+      if (onCollapse) {
+        onCollapse();
+      }
+    };
+
+    return (
+      <OutsideClickHandler onOutsideClick={onOutsideClick}>
+        <li
+          {...props}
+          className={classNames(
+            styles.Wrapper,
+            clone && styles.clone,
```

Comment:
className={classNames(
            styles.Wrapper,
            {
            [styles.clone]: clone
            })
 when we want to add classes by condition with boolean we use this like an object with class like a key and boolean like a value
 when we want to add class value, for example:
 props buttonType = text | contained
 <Button className={classNames([styles.[buttonType]])}>            
replace with such a method everywhere